ui/src/app/components/admin/audit-card-list-view.tsx

```diff
@@ -193,17 +210,115 @@ const AuditActionCard = (props: { action: AuditAction }) => {
   );
 };
 
-export const AuditActionCardListView = (props: { actions: AuditAction[]}) => {
+interface FilterEntry {
+  displayName?: string;
+  isActive: boolean;
+}
+
+const ActionTypeFilter = (props: {
+  activeActionTypes: { [key: string]: FilterEntry },
+  updateFilter: (actionType: string, isActive: boolean) => void }) => {
+
+  const {activeActionTypes, updateFilter} = props;
+
+  const toggleSelectedAction = (actionType) => {
+    updateFilter(actionType, !fp.get(`${actionType}.isActive`)(activeActionTypes));
+  };
+
+  return (
+      <React.Fragment>
+        <div style={{
+          margin: '0',
+          display: 'flex',
+          flexDirection: 'column',
+          textAlign: 'left',
+          border: '1px solid',
+          borderColor: colors.secondary,
+          width: 'fit-content'
+        }}>
+        <div style={{fontWeight: 600, color: colors.accent}}>Action Types</div>
+        {fp.entries(activeActionTypes).map((actionTypeValuePair: [string, FilterEntry], index) =>
+          <CheckBox
+            key={index}
+            id={`${actionTypeValuePair[0]}_active`}
+            onChange={() => toggleSelectedAction(actionTypeValuePair[0])}
+            label={actionTypeValuePair[1].displayName}
+            checked={actionTypeValuePair[1].isActive}
+            style={{margin: '0.25rem'}}/>)}
+        </div>
+    </React.Fragment>);
+};
+
+export const AuditActionCardListView = (props: { actions:  AuditAction[]}) => {
   const {actions} = props;
+  const [actionTypes, setActionTypes] = useState([]);
+  const [activeActionTypes, setActiveActionTypes] = useState({});
+  const [actionTypeToCardCount, setActionTypeToCardCount] = useState({});
+
+  useEffect(() => {
```

Comment:
Given that this effect runs on the prop change, and there is only one prop, it may be possible to eliminate all of the useEffect code and just compute all of the needed values. This would eliminate all 'state' in the react sense.